packages/grafana-sql/src/datasource/SqlDatasource.ts

```diff
@@ -78,9 +260,17 @@ export abstract class SqlDatasource extends DataSourceWithBackend<SQLQuery, SQLO
   interpolateVariable = (value: string | string[] | number, variable: VariableWithMultiSupport) => {
     if (typeof value === 'string') {
       if (variable.multi || variable.includeAll) {
-        return this.getQueryModel().quoteLiteral(value);
+        return this.getQueryModel().quoteLiteral(value); // add quotes `'`
       } else {
-        return String(value).replace(/'/g, "''");
+        // fixbug, the value will doesn't contains quotes `'` when checkbox of Multi value is not checked.
+        // Another the type of variable is scopeVar, has same promblem.
+        // Wrong Example:
+        //  select * from table where name in ($var_a) => select * from table where name in (a)
+        if (value === "''") {// PR: https://github.com/grafana/grafana/pull/56879
+          return String(value).replace(/'/g, "''");
+        } else {
+          return value
```

Comment:
This does not seem related to adding a log context and rather a potential bug fix, so it should be probably in separate PR.

Reply:
OK, I commit a another PR.